Deprecate cocotb.hook (#2201)
cocotb/cocotb: documentation/source/building.rst

    A comma-separated list of modules that should be executed before the first test.
    You can also use the :class:`cocotb.hook` decorator to mark a function to be run before test code.

    .. deprecated:: 1.5

>         :class:`cocotb.hook` has been deprecated and this variable is no longer used.

Reviewer: ```suggestion
        :class:`cocotb.hook` is deprecated, and in future this variable will be ignored.
```

Author: Forgot a `the` so I added it.
